app/Services/AuthorizesWithPhoenix.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Rogue\Services;
+
+trait AuthorizesWithPhoenix
+{
+    /**
+     * Run custom tasks before making a request.
+     *
+     * @see RestApiClient@raw
+     */
+    public function runAuthorizesWithPhoenixTasks($method, &$path, &$options, &$withAuthorization)
+    {
+        if ($withAuthorization) {
+            if (! isset($options['token'])) {
+                $authorizationHeader = [];
+                $authorizationHeader['X-CSRF-Token'] = $this->getAuthenticationToken();
```

Comment:
Oh, we should move these methods in here as well: `getAuthenticationToken` & `getAuthenticationCookie`. 🍪